components/layout/query.rs

```diff
@@ -639,18 +639,18 @@ pub fn process_node_scroll_area_request< N: LayoutNode>(requested_node: N, layou
             let bottom = max(iterator.union_rect.size.height, iterator.origin_rect.size.height);
             let left = max(iterator.union_rect.origin.x, iterator.origin_rect.origin.x);
             Rect::new(Point2D::new(left, iterator.origin_rect.origin.y),
-                      Size2D::new(iterator.origin_rect.size.width, bottom))
+                      Size2D::new(iterator.union_rect.size.width, bottom))
```

Comment:
So, this also includes left and top margins, right? That looks wrong per https://drafts.csswg.org/cssom-view/#scrolling-area, isn't it?